broadbean/sequence.py

```diff
@@ -478,21 +481,102 @@ def description(self):
+        """
+        with open(path_to_file, 'w') as fp:
+            json.dump(self.description, fp, indent=4)
+
+    @classmethod
+    def init_from_json(cls, path_to_file: str):  # -> 'Sequence':
+        """
+        Reads sequense from JSON file
+
+        Args: 
+            path_to_file: the path to the file to be read ex:
+            path_to_file/sequense.json
+            This function is the inverse of write_to_json
+            The JSON file needs to be structured as if it was writen
+            by the function write_to_json
+        """
+        new_instance = cls()
+        with open(path_to_file, 'r') as fp:
+            data_loaded = json.load(fp)  
+        knowfunctions = dict([('function PulseAtoms.{}'.format(fun),
+                               getattr(PulseAtoms, fun)) for fun in
+                             dir(PulseAtoms) if '__' not in fun])
+        awgspecs = data_loaded['awgspecs']
+        SR = awgspecs['SR']
+        ElemList = list(data_loaded.keys())
+
+        for Ele in ElemList[:-1]:
+            ChannelsList = list(data_loaded[Ele]['channels'].keys())
+            elem = Element()
+            for chan in ChannelsList:
+                SegMarlist = list(data_loaded[Ele]['channels'][chan].keys())
+                Seglist = [s for s in SegMarlist if 'segment' in s]
+                bp_sum = BluePrint()
+                i = 0
+                for Seg in Seglist:
+                    SegDict = data_loaded[Ele]['channels'][chan][Seg]
+                    bp_seg = BluePrint()
+                    bp_seg.setSR(SR)
+                    if SegDict['function'] == 'waituntil':
+                        arguments = data_loaded[Ele]['channels'][chan][Seg]['arguments'].values()
+                        arguments = (list(arguments)[0][0],)
+                        print(arguments)
+                        bp_seg.insertSegment(i, 'waituntil', arguments)
+                    else:
+                        arguments = tuple(data_loaded[Ele]['channels'][chan][Seg]['arguments'].values())
+                        bp_seg.insertSegment(i, knowfunctions[SegDict['function']],
+                                             arguments, name=re.sub(r'\d', "", SegDict['name']), dur=SegDict['durations'])
+                    i += 1
+                    bp_sum = bp_sum + bp_seg
+                bp_sum.marker1 = data_loaded[Ele]['channels'][chan]['marker1_abs']
+                bp_sum.marker2 = data_loaded[Ele]['channels'][chan]['marker2_abs']
+                listmarker1 = data_loaded[Ele]['channels'][chan]['marker1_rel']
+                listmarker2 = data_loaded[Ele]['channels'][chan]['marker2_rel']
+                bp_sum._segmark1 = [tuple(mark) for mark in listmarker1]
+                bp_sum._segmark2 = [tuple(mark) for mark in listmarker2]
+                bp_sum.setSR(SR)
+                elem.addBluePrint(int(chan), bp_sum)
+
+                ChannelAmplitude = awgspecs['channel{}_amplitude'.format(chan)]
```

Comment:
`channel_amplitude` might be a better name.

Reply:
I agree